lib/rucio/common/schema/__init__.py

```diff
@@ -0,0 +1,31 @@
+"""
+ Copyright European Organization for Nuclear Research (CERN)
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Authors:
+ - Vincent Garonne, <vincent.garonne@cern.ch>, 2017
+"""
+
+from ConfigParser import NoOptionError, NoSectionError
+
+from rucio.common import config
+
+if config.config_has_section('policy'):
+    try:
+        POLICY = config.config_get('policy', 'schema')
+    except (NoOptionError, NoSectionError) as error:
+        POLICY = 'generic'
+else:
+    POLICY = 'generic'
+
+if POLICY.lower() == 'generic':
+    from .generic import *  # NOQA pylint:disable=wildcard-import
+elif POLICY.lower() == 'atlas':
```

Comment:
Can't we do the import completely dynamic? I mean in this case we still basically need to maintain a list of all organisations we support.

Reply:
this is what i did for pilot2:


 workflow = __import__('pilot.workflow.%s' % args.workflow, globals(), locals(), [args.workflow], -1)
workflow.run(args)

so, whatever module name string you give to __import__, it will import with the correct python context

same logic can be applied here

Reply:
with policy loader and entry_point, this should be doable (for the next feature release)
unless you have some immediate suggestions ?

Reply:
setuptools can provide this. Example:
import pkg_resources
for impl in pkg_resources.iter_entry_points(
                    self.group, name):
                self.impls[name] = impl.load
                return impl.load()